governance/program/src/instruction.rs

```diff
@@ -0,0 +1,264 @@
+        instruction_end_index: u16,
+    },
+
+    /// [Requires Signatory token]
+    /// Remove Transaction from the Proposal
+    ///
+    ///   0. `[writable]` Proposal State account
+    ///   1. `[writable]` Proposal Transaction account
+    ///   2. `[writable]` Signatory account
+    ///   3. `[writable]` Signatory Validation account
+    ///   5. `[]` Proposal account
+    ///   6. `[]` Transfer Authority
+    ///   7. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_key]
+    ///   9. `[]` Token program account
+    RemoveTransaction,
+
+    /// [Requires Signatory token]
+    /// Update Transaction slot in the Proposal. Useful during reset periods
+    ///
+    ///   1. `[writable]` Proposal Transaction account
+    ///   2. `[writable]` Signatory account
+    ///   3. `[writable]` Signatory Validation account
+    ///   4. `[]` Proposal State account.
+    ///   5. `[]` Proposal account.
+    ///   6. `[]` Transfer authority.
+    ///   7. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_key]
+    ///   8. `[]` Token program account
+    UpdateTransactionDelaySlots {
+        /// On what slot this transaction slot will now run
+        delay_slots: u64,
+    },
+
+    /// [Requires Admin token]
+    /// Delete Proposal entirely.
+    ///
+    ///   0. `[writable]` Proposal state account pub key.
+    ///   1. `[writable]` Admin account
+    ///   2. `[writable]` Admin Validation account.
+    ///   3. `[]` Proposal account pub key.
+    ///   4. `[]` Transfer authority.
+    ///   5. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_key]
+    ///   6. `[]` Token program account.
+    DeleteProposal,
+
+    /// [Requires Signatory token]
+    /// Burns signatory token, indicating you approve of moving this Proposal from Draft state to Voting state.
+    /// The last Signatory token to be burned moves the state to Voting.
+    ///
+    ///   0. `[writable]` Proposal State account
+    ///   1. `[writable]` Signatory account
+    ///   2. `[writable]` Signatory Mint account
+    ///   3. `[]` Proposal account
+    ///   4. `[]` Transfer authority
+    ///   5. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_key]
+    ///   6. `[]` Token program account.
+    ///   7. `[]` Clock sysvar.
+    SignProposal,
+
+    /// [Requires Voting tokens]
+    /// Burns voting tokens, indicating you approve and/or disapprove of running this set of transactions. If you tip the consensus,
+    /// then the transactions can begin to be run at their time slots when people click execute. You are then given yes and/or no tokens
+    ///
+    ///   0. `[writable]` Governance voting record account
+    ///                   Can be uninitialized or initialized(if already used once in this proposal)
+    ///                   Must have address with PDA having seed tuple [Governance acct key, proposal key, your voting account key]
```

Comment:
would the PDA for this use your source token account address then? or would you use the owner of the token account?

Reply:
Voter (the owner of the source token account) + Proposal